experimental/Schemes/WeilDivisor.jl

```diff
@@ -0,0 +1,352 @@
+export WeilDivisor
+export scheme_type, ideal_sheaf_type, coefficient_ring_type, coefficient_type
+export scheme, components, coefficient_dict, coefficient_ring
+export in_linear_system
+
+export LinearSystem 
+export divisor, find_subsystem
+
+@Markdown.doc """
+    WeilDivisor
+
+A Weil divisor on a covered scheme ``X``; 
+stored as a formal linear combination over 
+some ring ``R`` of ideal sheaves on ``X``.
+"""
+@attributes mutable struct WeilDivisor{
+    CoveredSchemeType<:AbsCoveredScheme, 
+    CoefficientRingType<:AbstractAlgebra.Ring, 
+    CoefficientRingElemType<:AbstractAlgebra.RingElem
+   }
+  X::CoveredSchemeType # the parent
+  R::CoefficientRingType # the ring of coefficients
+  coefficients::IdDict{IdealSheaf, CoefficientRingElemType} # the formal linear combination
+
+  function WeilDivisor(
+      X::AbsCoveredScheme,
+      R::CoefficientRingType, 
+      coefficients::IdDict{<:IdealSheaf, CoefficientRingElemType};
+      check::Bool=true
+    ) where {CoefficientRingType, CoefficientRingElemType}
+    # TODO: Do we want to require that the different effective divisors 
+    # have the same underlying covering? Probably not.
+    if check
```

Comment:
These checks are so cheap that they should be done always. 
If `check` please  test that the ideals are radical ideals and of codimension 1.